src/api/use-subscription.ts

```diff
@@ -25,40 +28,64 @@ export function useSubscription({
   const [state, setState] = useState<SubscriptionState>('NOT_STARTED');
   const [webSocket, setWebSocket] = useState<WebSocket | null>(null);
 
+  const [reconnectCount, setReconnectCount] = useState(0);
+  const [retryCount, setRetryCount] = useState(0);
+  const exponentialBackoff = () => {
+    const delay = Math.pow(2, retryCount) * 1000;
+    return setTimeout(() => setRetryCount(retryCount + 1), delay);
+  };
+
   useEffect(() => {
+    let retryTimeout: NodeJS.Timeout | null = null;
     if (url) {
       const webSocket = new WebSocket(url);
+
       webSocket.onmessage = (event) => {
         setState(getSubscriptionStateCode(webSocket.readyState));
         onMessage && onMessage(event);
       };
+
       webSocket.onerror = (event) => {
         setState(getSubscriptionStateCode(webSocket.readyState));
         Bugsnag.notify(`WebSocket error "${event.message}" with url: ${url}`);
         onError && onError(event);
```

Comment:
Do we need to try to reconnect here?

Reply:
Looked into it now. It seem like when an `onerror` happens, an `onclose` event is also triggered, even if the connection hasn't been successfully opened in the first place. But an `onclose` event doesn't necessarily trigger `onerror`. If we add error handling here, we would need to add extra logic to make sure it isn't triggered twice, while the close event can handle both errors and "regular" close events quite easily.

So good question, but I think it's best to not add any reconnects here 😊